main/src/library/Condition.flix

```diff
@@ -40,7 +44,9 @@ namespace Condition {
         }
 
     ///
-    /// Causes the current thread to wait until it is signalled or interrupted, or the specified waiting time (nanos) elapses.
+    /// Causes the current thread to wait until it is signalled or interrupted, or the specified waiting
+    /// time(nanos) elapses. The method returns an estimate of the number of nanoseconds remaining to wait
+    /// given the supplied nanosTimeout value upon return, or a value less than or equal to zero if it timed out.
```

Comment:
Would it be clearer to return an enum from this? There seem to be two categories of return value: `Remaining(Int64)` and `TimedOut`.

Reply:
Possibly yes, the java class also has a simpler method which just returns a bool which is the same concept. I haven't used this for anything myself so I'm not sure what the best is.

Reply:
I am going to leave this as future work.